moveit_core/collision_detection/src/collision_tools.cpp

```diff
@@ -39,41 +39,44 @@
 
 namespace collision_detection
 {
-void getCostMarkers(visualization_msgs::MarkerArray& arr, const std::string& frame_id,
+void getCostMarkers(visualization_msgs::msg::MarkerArray& arr, const std::string& frame_id,
                     std::set<CostSource>& cost_sources)
 {
-  std_msgs::ColorRGBA color;
+  std_msgs::msg::ColorRGBA color;
   color.r = 1.0f;
   color.g = 0.5f;
   color.b = 0.0f;
   color.a = 0.4f;
-  getCostMarkers(arr, frame_id, cost_sources, color, ros::Duration(60.0));
+  getCostMarkers(arr, frame_id, cost_sources, color, rclcpp::Duration(60.0));
 }
 
-void getCollisionMarkersFromContacts(visualization_msgs::MarkerArray& arr, const std::string& frame_id,
+void getCollisionMarkersFromContacts(visualization_msgs::msg::MarkerArray& arr, const std::string& frame_id,
                                      const CollisionResult::ContactMap& con)
 {
-  std_msgs::ColorRGBA color;
+  std_msgs::msg::ColorRGBA color;
   color.r = 1.0f;
   color.g = 0.0f;
   color.b = 0.0f;
   color.a = 0.8f;
-  getCollisionMarkersFromContacts(arr, frame_id, con, color, ros::Duration(60.0));
+  getCollisionMarkersFromContacts(arr, frame_id, con, color, rclcpp::Duration(60.0));
 }
 
-void getCostMarkers(visualization_msgs::MarkerArray& arr, const std::string& frame_id,
-                    std::set<CostSource>& cost_sources, const std_msgs::ColorRGBA& color, const ros::Duration& lifetime)
+void getCostMarkers(visualization_msgs::msg::MarkerArray& arr, const std::string& frame_id,
+                    std::set<CostSource>& cost_sources, const std_msgs::msg::ColorRGBA& color,
+                    const rclcpp::Duration& lifetime)
 {
   int id = 0;
   for (const auto& cost_source : cost_sources)
   {
-    visualization_msgs::Marker mk;
-    mk.header.stamp = ros::Time::now();
+    rclcpp::Clock ros_clock(RCL_ROS_TIME);
```

Comment:
For what do we need ros_clock here? It could probably be used to initialize `mk.header.stamp` below by calling `ros_clock.now()` like already implemented in the changes below.

Reply:
done